src/goto-instrument/code_contracts.cpp

```diff
@@ -1542,11 +1542,38 @@ goto_programt assigns_clauset::havoc_code(
   goto_programt havoc_statements;
   for(assigns_clause_targett *target : targets)
   {
+    //    if(target != NULL) goto x;
+    //      havoc_statements
+    // z: skip
+
+    // create the z label
+    goto_programt tmp_z;
+    goto_programt::targett z = tmp_z.add(goto_programt::make_skip(location));
+
+    // TODO: generalize this. Currently only supports pointers
```

Comment:
Could you add a bit more explanation to this comment?

I am not sure what generalization is expected here. What needs to be checked for other `target_type`s: `Array` and `Struct`?

cc: @feliperodri

Reply:
We should
1. Remove the `TODO` comment. We don't need to generalize this, it should work only for dereferences;
2. We should close the GitHub issue because of (1);
3. We should check for the expr id in the if statement.

Reply:
@feliperodri I think point 3 may not be necessary. The variables in the assigns clause are represented as `address_of` expressions which point to the corresponding symbol, dereference, etc... As a result, I think that any kind of if statement or invariant would not be necessary.